### ISSUE TITLE
[Feature Request] More examples

### ISSUE THREAD

**Issue by karandwivedi42** [pytorch/ignite#202]
It would be good to have more examples from the official [examples repo](https://github.com/pytorch/examples) reproduced in ignite.

Specifically, I would like to see if ignite can lead to compact but full-featured code in:

1. Imagenet
2. Reinforcement Learning
3. Neural Style

before I feel confident enough to use it.

**Comment by alykhantejani** [comment]
Thanks for the suggestion. I think more examples would be great and something we'd like to get to. Of course  PR's are welcome!

**Comment by vfdev-5** [comment]
@karandwivedi42 good proposition!

Personnally, I started some more examples [here](https://github.com/vfdev-5/ignite-examples) and find out that classification tasks can be easily implemented, so ImageNet example won't be very different from MNIST. 
Concerning detection or segmentation examples, we will need to rewrite all the stuff: data augmentation, data transformations (bounding box encoding/decoding) around the task depending on the architecture used, e.g. Faster-RCNN, RetinaNet for detection or a basic UNet for segmentation.

I do not know whether RL examples and Neural Style should suffer of the same "problem"...

**Comment by alykhantejani** [comment]
with regards to detection, I think the transforms etc. should live in `torchvision` or a detection specific package. Beyond that it's just a supervised training task I believe so don't see any difficulty of using ignite for this.

Although, I still think examples of how to use ignite for these would be cool

**Comment by vfdev-5** [comment]
@alykhantejani  I agree. We can create github project on this issue.

And maybe can port to ignite more examples from [pytorch/examples](https://github.com/pytorch/examples), where there are imagenet, RL...

**Comment by elanmart** [comment]
Language modelling example from `pytorch/examples` could use a refactor...

**Comment by jasonkriss** [comment]
Agreed that this sounds like a good idea. Neural Style, detection, Imagenet etc should work no problem. The only thing we may want is some more task-specific metrics. RL will probably not work super well, though it will depend on the algorithm used.

**Comment by vfdev-5** [comment]
In a previous issue users asked about hyperparameter tuning, maybe we could also provide an example using [ray library](http://ray.readthedocs.io/en/latest/tune.html). 
Upvote if you think this can be interesting.

**Comment by elanmart** [comment]
I'll add a language modelling example hopefully this week

**Comment by vfdev-5** [comment]
me too I was planning to provide an example on image segmentation task. Detection task would require a lot of side preparation stuff, however we could port something from roytseng/Detectron.pytorch or wait when @fmassa will release official Detectron Pytorch and we could simplify the training part with Ignite :)

**Comment by anmolsjoshi** [comment]
@vfdev-5 if you want to collaborate on a detection task, I'm happy to help out. I have an implementation of U-Net (segmentation) that I can port over to ignite over the next week.

**Comment by rpatrik96** [comment]
Hi everyone!

I hope it is the right place to ask: I am really fascinated with Ignite and I would like to make my contribution to it, is there anything I could help?
I am currently doing some transfer learning stuff with Resnet, what is your take, would it be appropriate to include a tramsfer learning example in the Ignite repo? If yes, I am happy to write it :).

**Comment by alykhantejani** [comment]
Hi @rpatrik96, 

I'm not sure that transfer learning for classification is a useful example at the moment, mostly because the changes come in just how you load your model and set your learning rate, and doesn't necessarily showcase ignite any differently than the classification example does. 

However, if you'd like to contribute more we welcome that :) please have a look at the issues, especially the ones tagged "help-wanted" and "good first issue" are good starting points.

**Comment by rave78** [comment]
Hi everyone!

I was wondering if it is ok to update the mnist example (mnist.py) with tqdm (two progress bars -- one for epoch and the other for iteration).

**Comment by vfdev-5** [comment]
Hi @rave78 yes, that can be interesting to integrate tqdm into our basic examples like mnist. So, please send a PR on that and we can iterate over it. 
Maybe only I progress bar on batches during the training would of interest, IMO. But let's see in the PR...

**Comment by philip-bl** [comment]
I can create a tensorboard logger example of a very simple image segmentation task, where the model finds which pixels represent borders of some cells. I doesn't use much ignite functionality absent from other example, but I think overall the are not enough examples of using tensorboard logger. Will that be useful? Would you like me to do it? Below you can see some images and the corresponding targets.
![some_images](https://user-images.githubusercontent.com/6838251/57180140-6460fd00-6e8e-11e9-9888-6c6c1f27b953.jpg)

**Comment by vfdev-5** [comment]
@philip-bl sure that ignite lacks examples on segmenation/object detection tasks. An example on [Pascal VOC Segmentation from torchvision](https://github.com/pytorch/vision/blob/master/torchvision/datasets/voc.py) can be a more canonical, but we can use something else. What is the dataset you are using ?

**Comment by philip-bl** [comment]
@vfdev-5 [bbbc018_v1](https://data.broadinstitute.org/bbbc/BBBC018/)

**Comment by vfdev-5** [comment]
@philip-bl, ok, go ahead, send a PR and then let's discuss how to make it useful. You can provide a script example or notebook, as you like

**Comment by vfdev-5** [comment]
Closed the issue as we provided most of discussed examples
